storage/replica.go

```diff
@@ -964,11 +969,9 @@ func (r *Replica) addWriteCmd(ctx context.Context, ba roachpb.BatchRequest, wg *
 		case <-ctx.Done():
 			return br, roachpb.NewError(ctx.Err())
```

Comment:
Done.

And yeah the structure was weird because of the mutex. Moving the lease acquisition up allowed it to be a lot cleaner.